### DISCUSSION TITLE
Connections to Redis time out

### DISCUSSION THREAD

**Discussion by uhthomas** [renovatebot/renovate#20294] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.116.1

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

N/A

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Connections to Redis using the [Helm chart](https://github.com/renovatebot/helm-charts) time out (see debug logs section).

The `RENOVATE_REDIS_URL` environment variable is set to `redis://renovate-redis-master`.

I can connect to Redis if I port-forward the service.

```sh
❯ k -n renovate port-forward svc/renovate-redis-master 6379
Forwarding from 127.0.0.1:6379 -> 6379
Forwarding from [::1]:6379 -> 6379
```

```sh
❯ redis-cli PING
PONG
```

I found https://github.com/redis/node-redis/issues/1656... Could be relevant? 

### Relevant debug logs

<details><summary>Logs</summary>

```
{"name":"renovate","hostname":"renovate-y-27930060-sxr5v","pid":4,"level":60,"logContext":"hdEDQh6klgoduRWeMTG__","err":{"message":"Connection timeout","stack":"Error: Connection timeout\n    at Socket.<anonymous> (/opt/buildpack/tools/renovate/34.116.1/node_modules/@redis/client/dist/lib/client/socket.js:164:124)\n    at Object.onceWrapper (node:events:627:28)\n    at Socket.emit (node:events:513:28)\n    at Socket.emit (node:domain:489:12)\n    at Socket._onTimeout (node:net:550:8)\n    at listOnTimeout (node:internal/timers:559:17)\n    at processTimers (node:internal/timers:502:7)"},"msg":"Fatal error: Connection timeout","time":"2023-02-07T21:02:46.333Z","v":0}
{"name":"renovate","hostname":"renovate-y-27930060-sxr5v","pid":4,"level":20,"logContext":"hdEDQh6klgoduRWeMTG__","msg":"Renovate exiting","time":"2023-02-07T21:02:46.334Z","v":0}
{"name":"renovate","hostname":"renovate-y-27930060-sxr5v","pid":4,"level":30,"logContext":"hdEDQh6klgoduRWeMTG__","loggerErrors":[{"name":"renovate","level":60,"logContext":"hdEDQh6klgoduRWeMTG__","err":{"message":"Connection timeout","stack":"Error: Connection timeout\n    at Socket.<anonymous> (/opt/buildpack/tools/renovate/34.116.1/node_modules/@redis/client/dist/lib/client/socket.js:164:124)\n    at Object.onceWrapper (node:events:627:28)\n    at Socket.emit (node:events:513:28)\n    at Socket.emit (node:domain:489:12)\n    at Socket._onTimeout (node:net:550:8)\n    at listOnTimeout (node:internal/timers:559:17)\n    at processTimers (node:internal/timers:502:7)"},"msg":"Fatal error: Connection timeout"}],"msg":"Renovate is exiting with a non-zero code due to the following logged errors","time":"2023-02-07T21:02:46.334Z","v":0}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
please add the port to the connection string

**Comment by uhthomas** [comment]
I tried that and it had the same result. I also tried adding a `/0` at the end, again with the same result.

**Comment by viceice** [comment]
I'm using redis with renovate for a long time and never had those issue. so it's probably an issue in your environment.

i think you need to ensure that redis is fully running before starting renovate.

btw: I'm using keydb as a drop-in replacement which supports multi master redis deployment.

**Comment by rarkins** [comment]
Converted to discussion

**Comment by uhthomas** [comment, marked as answer]
You're absolutely right, thank you.

This was caused by a restrictive NetworkPolicy in Kubernetes. I verified that Renovate was not the issue as `redis-cli` was also not able to connect.

```
ubuntu@renovate-a-27931140-xghl4:~/redis-7.0.8$ src/redis-cli -h renovate-redis-master
Could not connect to Redis at renovate-redis-master:6379: Connection timed out
not connected> exit
```

After adjusting the NetworkPolicy, I could connect from within the Renovate pod.

```
ubuntu@renovate-a-27931140-xghl4:~/redis-7.0.8$ src/redis-cli -h renovate-redis-master
renovate-redis-master:6379> exit
```

Thanks for the help! Sorry for the trouble.
